test/js/ez-asynchronous-block.js

```diff
@@ -79,14 +76,51 @@ describe('ez-asynchronous-block', function() {
         });
 
         it('should update the content', function (done) {
+            sinon.spy(window, 'fetch');
             element.addEventListener('ez:asynchronousBlock:updated', function () {
                 assert.isNotNull(element.querySelector('.updated'));
                 done();
             });
             element.load();
         });
 
+        describe('redirection', function () {
+            const url = 'h/e/r/e';
+
+            beforeEach(function () {
+                const headers = {
+                    get: function (param) {
+                        if ( param === 'App-Location') {
+                            return url;
+                        }
+                    },
+                };
+                const response =  {'status': 205, 'headers': headers};
+
+                sinon.stub(window, 'fetch', function () {
+                    return Promise.resolve(response);
+                });
+            });
+
+            it('should be redirected after 205 status', function (done) {
+                const load = function (e) {
+                    assert.isTrue(element.loaded);
+                    assert.equal(e.detail.url, url, 'The event should contains the url');
+                    document.removeEventListener('ez:navigateTo', load);
```

Comment:
you should put that first otherwise if one of the assertion above fails, the clean up won't be done